### PR TITLE
Feature/icky 452 checkout error validation

### PR DESCRIPTION
**Pull request by kibo-amolpawar**

### CODE REVIEW

File: pages/checkout.vue [hunk 1]
```diff
@@ -390,19 +353,21 @@ export default {
           billingContact: { ...updatedBillingAddress.value, email: personalDetails.value?.email },
         },
       }
-      await setBillingInfo(params)
+      await setBillingInfo(params).then(async () => {
```
Comment (kibo-kevinwatts) [inline pages/checkout.vue]: always use async/await, avoid "then" promises

File: composables/useUserAddresses.ts [hunk 2]
```diff
@@ -6,21 +6,21 @@ import {
...
   UpdateUserAddressParams,
   DeleteUserAddressParams,
 } from "@/server/types/Api"
+import type { Maybe, CustomerContactCollection } from "@/server/types/GraphQL"
 
 export const useUserAddresses = () => {
   const nuxt = useNuxtApp()
   const fetcher = nuxt.nuxt2Context.$gqlFetch
-  const userShippingAddresses = useState<Maybe<Array<CustomerContact>>>(
+  const userShippingAddresses = useState<Maybe<Array<CustomerContactCollection>>>(
```
Comment (kibo-kevinwatts) [inline composables/useUserAddresses.ts]: this type should stay CustomerContact, may need to rebase branch

File: components/KiboAddressForm.vue [hunk 3]
```diff
@@ -137,9 +177,88 @@ export default {
...
+      schema
+        .validate(props.value)
+        .then(() => {
+          context.emit("validateForm", true)
+        })
+        .catch(() => {
+          context.emit("validateForm", false)
+        })
+    }
+    const schema = yup.object({
```
Comment (kibo-kevinwatts) [inline components/KiboAddressForm.vue]: can we move schema definition and validate outside of the component?
How can we write unit tests for these validations here?

File: components/KiboAddressForm.vue [hunk 4]
```diff
@@ -137,9 +177,88 @@ export default {
...
+      if (field === "phoneNumbers") {
+        validateField = `${field}.home`
+      } else if (address.includes(field)) {
+        validateField = `address[${field}]`
+      } else {
+        validateField = field
+      }
+      schema
+        .validateAt(validateField, addressDetails.value)
+        .then(() => {
```
Comment (kibo-kevinwatts) [inline components/KiboAddressForm.vue]: always use async/await for promises

File: pages/checkout.vue [hunk 5]
```diff
@@ -275,16 +228,16 @@ export default {
...
     // const applyPromocode = async (couponApplied) => await applyCoupon(couponApplied)
 
-    const isValidCoupon = computed(() => !getOrder.value?.invalidCoupons[0]?.couponCode)
+    // const isValidCoupon = computed(() => !getOrder.value?.invalidCoupons[0]?.couponCode)
 
-    const invalidCouponErrorText = computed(
-      () =>
-        `${getOrder.value?.invalidCoupons[0]?.couponCode} ${context.root.$t("is an invalid code")}`
-    )
+    // const invalidCouponErrorText = computed(
```
Comment (kibo-kevinwatts) [inline pages/checkout.vue]: Delete the code if uneeded, don't comment out and leave

**Review by kibo-kevinwatts — CHANGES_REQUESTED** [review]

File: components/KiboAddressForm.vue [hunk 6]
```diff
@@ -6,65 +6,99 @@
         name="firstName"
         class="form__element form__element--half"
         required
+        :valid="!errors.firstName"
+        :error-message="errors.firstName"
         :value="addressDetails.firstName"
         @input="updateField('firstName', $event)"
+        @blur="validate('firstName')"
+        @keypress="validate('firstName')"
```
Comment (TriptiVermaXP) [inline components/KiboAddressForm.vue]: Do we need to validate the value on _keypress_ from everywhere expect password, if validating on _blur_? 
Performance issue may occur if validating on every input value entered.

File: components/KiboAddressForm.vue [hunk 7]
```diff
@@ -137,9 +177,88 @@ export default {
...
+          .string()
+          .required(context.root.$t("Required"))
+          .matches(/^[0-9]+$/, "Must be only digits"),
+        countryCode: yup.string().required(context.root.$t("Required")),
+      }),
+      phoneNumbers: yup.object({
+        home: yup
+          .string()
+          .required(context.root.$t("Required"))
+          .matches(/^[+]*[(]{0,1}[0-9]{1,3}[)]{0,1}[-\s\\./0-9]*$/g, "Must be valid phone number"),
```
Comment (TriptiVermaXP) [inline components/KiboAddressForm.vue]: Can't we move validations on useUiValidation.ts to keep all validations on one place?

File: pages/checkout.vue [hunk 8]
```diff
@@ -505,32 +471,32 @@ export default {
...
+      //     title: "Easy shipping",
+      //     description: "You’ll receive dispatch confirmation and an arrival date",
+      //     icon: "shipping",
+      //   },
+      //   {
+      //     title: "Changed your mind?",
+      //     description: "Rest assured, we offer free returns within 30 days",
+      //     icon: "return",
+      //   },
+      // ],
```
Comment (TriptiVermaXP) [inline pages/checkout.vue]: Please remove the comment code.

**Review by TriptiVermaXP — CHANGES_REQUESTED** [review]
Please check review comments.

File: pages/checkout.vue (hunk 8 above)
Comment (kibo-amolpawar) [reply pages/checkout.vue]: Resolved.

File: components/KiboAddressForm.vue (hunk 7 above)
Comment (kibo-amolpawar) [reply components/KiboAddressForm.vue]: Resolved.

File: components/KiboAddressForm.vue (hunk 6 above)
Comment (kibo-amolpawar) [reply components/KiboAddressForm.vue]: Resolved.

File: pages/checkout.vue (hunk 5 above)
Comment (kibo-amolpawar) [reply pages/checkout.vue]: Resolved.

File: components/KiboAddressForm.vue (hunk 4 above)
Comment (kibo-amolpawar) [reply components/KiboAddressForm.vue]: Resolved.

File: components/KiboAddressForm.vue (hunk 3 above)
Comment (kibo-amolpawar) [reply components/KiboAddressForm.vue]: Resolved.

File: composables/useUserAddresses.ts (hunk 2 above)
Comment (kibo-amolpawar) [reply composables/useUserAddresses.ts]: Resolved.

File: pages/checkout.vue (hunk 1 above)
Comment (kibo-amolpawar) [reply pages/checkout.vue]: Resolved.

**Review by kibo-chandradeeptalaha — APPROVED** [review]

**Pull request merged**